services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/NeoBolt.java

```diff
@@ -0,0 +1,210 @@
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.nbworker.bolts;
+
+import com.google.common.collect.Lists;
+import org.apache.storm.task.OutputCollector;
+import org.apache.storm.task.TopologyContext;
+import org.apache.storm.topology.OutputFieldsDeclarer;
+import org.apache.storm.topology.base.BaseRichBolt;
+import org.apache.storm.tuple.Fields;
+import org.apache.storm.tuple.Tuple;
+import org.neo4j.driver.v1.AccessMode;
+import org.neo4j.driver.v1.AuthTokens;
+import org.neo4j.driver.v1.Driver;
+import org.neo4j.driver.v1.GraphDatabase;
+import org.neo4j.driver.v1.Record;
+import org.neo4j.driver.v1.Session;
+import org.neo4j.driver.v1.StatementResult;
+import org.neo4j.driver.v1.Value;
+import org.openkilda.messaging.info.event.IslChangeType;
+import org.openkilda.messaging.info.event.IslInfoData;
+import org.openkilda.messaging.info.event.PathNode;
+import org.openkilda.messaging.info.event.SwitchInfoData;
+import org.openkilda.messaging.info.event.SwitchState;
+import org.openkilda.messaging.nbtopology.request.GetLinksRequest;
+import org.openkilda.messaging.nbtopology.request.GetSwitchesRequest;
+import org.openkilda.messaging.nbtopology.request.ReadDataRequest;
+import org.openkilda.wfm.topology.nbworker.StreamType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class NeoBolt extends BaseRichBolt {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NeoBolt.class);
+
+    private Driver driver;
+
+    private String address;
+    private String login;
+    private String password;
+
+    private OutputCollector outputCollector;
+
+    public NeoBolt(String host, String login, String password) {
+        this.address = String.format("bolt://%s", host);
+        this.login = login;
+        this.password = password;
+    }
+
+    @Override
+    public void prepare(Map stormConf, TopologyContext context, OutputCollector collector) {
+        LOGGER.info("Connecting to neo4j server {}", address);
+        driver = GraphDatabase.driver(address, AuthTokens.basic(login, password));
+        LOGGER.info("NEO4J connected {} (login=\"{}\")", address, login);
+        this.outputCollector = collector;
+    }
+
+    @Override
+    public void execute(Tuple input) {
+        final String streamId = input.getSourceStreamId();
+        final String correlationId = input.getString(1);
+        LOGGER.debug("Received request with correlationId: {}", correlationId);
+
+        try {
+            switch (StreamType.valueOf(streamId)) {
+                case READ:
```

Comment:
as for me - this incorrect level of requests split. Requests should be splitted based on type of resources(ISL, Switch, Flow, etc) not kind of operation. Because most(or even all) operations in this topology will be "read"... it will increase size of this bolt to enormous values.